pkg/operator/ceph/object/objectstore.go

```diff
@@ -405,19 +405,32 @@ func createMultisite(objContext *Context, endpointArg string) error {
 			if err != nil {
 				return errorOrIsNotFound(err, "failed to create ceph zone %q, for reason %q", objContext.Zone, output)
 			}
-			logger.Debugf("created zone %v", objContext.Zone)
+			logger.Debugf("created zone %q", objContext.Zone)
 		} else {
-			return errorOrIsNotFound(err, "radosgw-admin zone get failed with code %d, for reason %q", strconv.Itoa(code), output)
+			return errorOrIsNotFound(err, "'radosgw-admin zone get' failed with code %d, for reason %q", strconv.Itoa(code), output)
+		}
+	}
+
+	// check if the period exists
+	output, err = runAdminCommand(objContext, false, "period", "get")
+	if err != nil {
+		code, err := exec.ExtractExitCode(err)
+		// ENOENT means “No such file or directory”
+		if err == nil && code == int(syscall.ENOENT) {
+			// period does not exist and so needs to be created
+			updatePeriod = true
```

Comment:
If the period doesn't exist, won't `updatePeriod` already be set previously since the zone, zonegroup, and realm were also created? Or you're guarding against if the operator dies in the middle of the reconcile?

Reply:
Hmm... that's a good question. I assume `get` will fail if it wasn't set, but I can't remember if that's what I found out from my testing or if that's an assumption... let me check on that.

Reply:
I verified that if `period update --commit` hasn't been called, then `period get` will fail. Also verified in a manual test that the period is updated on the first reconcile and not on subsequent reconciles.

Reply:
AFAIR in the https://bugzilla.redhat.com/show_bug.cgi?id=2002220 setup, the `radosgw-admin period get` was working fine with containing only realm data but `radowgw-admin period get --staging` was failing with ENOENT
```
period init failed: (2) No such file or directory
2021-09-28T05:49:50.508+0000 7fd753369380  0 failed reading obj info from .rgw.root:periods.3f5441b6-b2ea-43fe-a64e-8cb78e5fe8ae:staging: (2) No such file or directory
```
So IMO it is better to add `--staging` for the `radosgw-admin period get` command from this experience

Reply:
This implementation avoids using `--staging` at all, but that feedback was useful for adding to this issue: https://github.com/rook/rook/issues/8879